pkg/testcase/restartservice.go

```diff
@@ -0,0 +1,34 @@
+package testcase
+
+import (
+	"time"
+
+	"github.com/rancher/distros-test-framework/factory"
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestRestartService() {
+	delay := time.After(30 * time.Second)
+	product, err := shared.GetProduct()
+	Expect(err).NotTo(HaveOccurred(), "failed to get product")
+	c := factory.AddCluster(GinkgoT())
+
+	for _, ip := range c.ServerIPs {
+		cmd, err := GetSystemCtlCmd(product, "restart", "server")
```

Comment:
we could directly do the shared.ManageService  call - which does the getsytemctl+run restart cmd, for you - it ll get the systemctl cmd and run the restart service as well. any errors will be caught and logged internally as well. 
```
// ManageService action:stop/start/restart/status product:rke2/k3s ips:ips array for nodeType:agent/server
func ManageService(product, action, nodeType string, ips []string) (string, error) {
	for _, ip := range ips {
		cmd, getError := GetSystemCtlCmd(product, action, nodeType)
		if getError != nil {
			return ip, getError
		}
		_, err := RunCommandOnNode(cmd, ip)
		if err != nil {
			return ip, err
		}
	}

	return "", nil
}
```

Reply:
truee that ! tk you